### PR TITLE
Handle the error when user request an invalid id

### PR DESCRIPTION
**Pull request by brunoocasali**
Context: https://github.com/meilisearch/meilisearch/issues/2555

Even keeping the u32 for performance reasons, the user still can reach a dead end by accessing a route with a big id, if they do that, it will respond with a 404 + internal error with an incorrect content-type.

This PR provides a way to handle this kind of error.

Code tips are appreciated 😬

### CODE REVIEW

File: meilisearch-lib/src/tasks/error.rs [hunk 1]
```diff
@@ -14,6 +14,8 @@ pub enum TaskError {
     UnexistingTask(TaskId),
     #[error("Internal error: {0}")]
     Internal(Box<dyn std::error::Error + Send + Sync + 'static>),
+    #[error("Task id is out of bounds.")]
+    OutOfBounds(),
```
Comment (MarinPostma) [inline meilisearch-lib/src/tasks/error.rs:18]: no parenthesis is necessary here
```suggestion
    OutOfBounds,
```

File: meilisearch-lib/src/tasks/error.rs [hunk 2]
```diff
@@ -29,6 +31,7 @@ impl ErrorCode for TaskError {
         match self {
             TaskError::UnexistingTask(_) => Code::TaskNotFound,
             TaskError::Internal(_) => Code::Internal,
+            TaskError::OutOfBounds() => Code::TaskNotFound,
```
Comment (MarinPostma) [inline meilisearch-lib/src/tasks/error.rs:34]:
```suggestion
            TaskError::OutOfBounds => Code::TaskNotFound,
```

File: meilisearch-http/src/routes/tasks.rs [hunk 3]
```diff
@@ -190,10 +191,9 @@ async fn get_task(
         Some(filters)
     };
 
-    let task: TaskView = meilisearch
-        .get_task(task_id.into_inner(), filters)
-        .await?
-        .into();
+    let id = task_id.ok_or(TaskError::OutOfBounds())?;
```
Comment (MarinPostma) [inline meilisearch-http/src/routes/tasks.rs:194]:
```suggestion
    let id = task_id.ok_or(TaskError::OutOfBounds)?;
```

File: meilisearch-http/src/routes/tasks.rs [hunk 4]
```diff
@@ -169,7 +170,7 @@ async fn get_tasks(
 
 async fn get_task(
     meilisearch: GuardedData<ActionPolicy<{ actions::TASKS_GET }>, MeiliSearch>,
-    task_id: web::Path<TaskId>,
+    task_id: Option<web::Path<TaskId>>,
```
Comment (MarinPostma) [inline meilisearch-http/src/routes/tasks.rs:173]: this is actually very strange. It works, but it's more like a ugly hack. If you really want to catch any kind of overflow, we will want to parse it to a `String` first, and then try to parse a `TaskId` from that

```suggestion
task_id: web::Path<String>,
```

File: meilisearch-http/src/routes/tasks.rs (hunk 3 above)
Comment (MarinPostma) [inline meilisearch-http/src/routes/tasks.rs:194]:
```suggestion
    let id = task_id.parse::<TaskId>().map_err(|_| TaskError::OutOfBounds)?;
```

File: meilisearch-http/src/routes/tasks.rs (hunk 4 above)
Comment (brunoocasali) [reply meilisearch-http/src/routes/tasks.rs:173]: I learned this [here](https://github.com/actix/actix-web/discussions/1986) 🤔

**Comment by Kerollmops** [comment]
I am closing this in favor of #2593, thank you for your contribution @brunoocasali and your review @MarinPostma 🙏

**Pull request closed without merging**